### DISCUSSION TITLE
Allow easier creation of details blocks

### DISCUSSION THREAD

**Discussion by Martmists-GH** [community/community#18799] [Projects and Issues]
GFM already differs from Markdown, so it shouldn't be too out-of-the-ordinary to add new elements that don't exist in Markdown.

Currently, to make a details block for logs, source files, errors, etc. I have to write the following:
```html
<details>
<!-- I can't use `` here for some reason -->
<summary><code>my_file.txt</code></summary>  
<!-- Similarly I can't use ``` here, and it won't have syntax highlighting -->
<!-- I have to use pre to make indents work as expected, but that also means I can't use newlines between the code tags without noticing the blank lines in the result -->
<pre><code>line 1
line 2
line 3</code></pre>
</details>
```
Resulting in:
<details>

<summary><code>my_file.txt</code></summary>  

<pre><code>line 1
line 2
line 3</code></pre>
</details>

---

I feel like there should be a much easier way to declare these, though I don't have any recommendations on how to approach it.

**Comment by aliuq** [comment]
Agree it, Long code should be collapsed
